velox/connectors/hive/iceberg/tests/TransformE2ETest.cpp

```diff
@@ -0,0 +1,646 @@
+
+    for (const auto& dir : firstLevelDirs) {
+      if (std::filesystem::is_directory(dir)) {
+        allPartitionDirs.push_back(
+            std::filesystem::path(dir).filename().string());
+        auto subDirs = listDirectoriesRecursively(dir);
+        allPartitionDirs.insert(
+            allPartitionDirs.end(), subDirs.begin(), subDirs.end());
+      }
+    }
+
+    return allPartitionDirs;
+  }
+
+  static void verifyPartitionCount(
+      const std::string& outputPath,
+      int32_t expectedPartitionCount) {
+    const auto partitionDirs = firstLevelDirectories(outputPath);
+
+    ASSERT_EQ(partitionDirs.size(), expectedPartitionCount)
+        << "Expected " << expectedPartitionCount << " partitions, got "
+        << partitionDirs.size();
+
+    for (const auto& dir : partitionDirs) {
+      const auto dirName = std::filesystem::path(dir).filename().string();
+      ASSERT_TRUE(dirName.find('=') != std::string::npos)
+          << "Partition directory " << dirName
+          << " does not follow Iceberg naming convention";
+    }
+  }
+
+  // Verify the total row count across all partitions.
+  void verifyTotalRowCount(
+      const RowTypePtr& rowType,
+      const std::string& outputPath,
+      int32_t expectedRowCount) {
+    auto splits = createSplitsForDirectory(outputPath);
+
+    const auto plan = PlanBuilder().tableScan(rowType).planNode();
+
+    const auto actualRowCount =
+        AssertQueryBuilder(plan).splits(splits).runWithoutResults();
+
+    ASSERT_EQ(actualRowCount, expectedRowCount);
+  }
+
+  // Verify data in a specific partition.
+  void verifyPartitionData(
+      const RowTypePtr& rowType,
+      const std::string& partitionPath,
+      const std::string& partitionFilter,
+      const int32_t expectedRowCount) {
+    auto splits = createSplitsForDirectory(partitionPath);
+
+    auto countPlan = PlanBuilder().tableScan(rowType).planNode();
+
+    const auto actualRowCount =
+        AssertQueryBuilder(countPlan).splits(splits).runWithoutResults();
+
+    ASSERT_EQ(actualRowCount, expectedRowCount);
+
+    const auto dataPlan =
+        PlanBuilder().tableScan(rowType).filter(partitionFilter).planNode();
+    const auto filteredRowCount =
+        AssertQueryBuilder(dataPlan).splits(splits).runWithoutResults();
+    ASSERT_EQ(expectedRowCount, filteredRowCount);
+  }
+
+  static std::pair<std::string, std::string> parsePartitionDirName(
+      const std::string& partitionDir) {
+    auto eq = partitionDir.find('=');
+    auto us = partitionDir.rfind('_', eq - 1);
+    auto columnName = partitionDir.substr(0, us);
+    auto value = partitionDir.substr(eq + 1);
+    return {columnName, value};
+  }
+
+  static int32_t computeBucketHash(
+      const StringView& value,
+      int32_t numBuckets) {
+    int32_t hash = functions::iceberg::Murmur3Hash32::hashBytes(
+        value.data(), value.size());
+    return ((hash & 0x7FFFFFFF) % numBuckets);
+  }
+};
+
+TEST_F(TransformE2ETest, identity) {
```

Comment:
This test seems to be verifies the behavior when each row becomes a separate partition. This doesn't seem to be a common case. Why did you choose this particular scenario?